pyseq.py

```diff
@@ -128,6 +139,24 @@ def __init__(self, item):
         self.head = self.name
         self.tail = ''
 
+    def __eq__(self, other):
```

Comment:
I think test_walk_succeed was the unit test that was having the issues, I'll having to double check that though.